src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -676,6 +676,20 @@ public static async ValueTask<FormulaValue> SortTable(EvalVisitor runner, EvalVi
             }
         }
 
+        public static FormulaValue TableConcatenate(IRContext irContext, FormulaValue[] args)
+        {
+            var tables = Array.ConvertAll(
+                args,
+                arg => arg switch
+                {
+                    TableValue r => r.Rows,
+                    BlankValue b => new List<DValue<RecordValue>> { DValue<RecordValue>.Of(b) },
+                    _ => new List<DValue<RecordValue>> { DValue<RecordValue>.Of((ErrorValue)arg) },

```

Comment:
>(ErrorValue) [](http://example.com/codeflow?start=80&length=12)

If you know this is going to be an ErrorValue, can you match it in the switch statement? And have the `_` "else" clause throw an exception (since it should never be hit)?